### ISSUE TITLE
Tauri-Plugin-Sql: BEGIN EXCLUSIVE TRANSACTION not working as expected

### ISSUE THREAD

**Issue by fmabap** [tauri-apps/plugins-workspace#1817]
Hello,
It’s my first project with Tauri and my first project where I use a SQL Lite DB.

I’m using this plugin in version 1 and I have the following issue:

The program is a single user application, and I have only one DB connection that is the whole time during the run of the program open.
When the user clicks on the save button, then I call await db.execute("BEGIN EXCLUSIVE TRANSACTION;");

I do then some selects where I call db.execute again and then I insert or update some data in multiple tables with some additional calls of db.execute. I call at the end db.execute(COMMIT;), or db.execute(ROLLBACK WORK;) if there was an error. Each db.execute is called with await.

When the user is in a special input field and clicks then directly on a save button, then the blur event of the field is raised where some selects will be executed in parallel. 

The issue is, that the “BEGIN EXCLUSIVE TRANSACTION” is not raising an exception, that the DB is locked, but 2 or 3 db.executes later I get sometimes (not always) the error that the DB is locked after a time of 5 – 10 seconds.

Is this a bug or must all actions that belongs to the transaction be handled in one db.execute call (this is not possible in my scenario)?

Best regards
Frank

**Comment by FabianLars** [comment]
hmm, transactions were never really looked at explicitly, so i think this counts as a duplicate of https://github.com/tauri-apps/plugins-workspace/issues/886.
